c10/core/TensorImpl.h

```diff
@@ -1341,6 +1342,14 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     device_opt_ = storage_.device();
   }
 
+  virtual void set_memory_format_tag(MemoryFormat memory_format);
+
+  virtual void reset_memory_format_tag() {
+    memory_format_tag_ = MemoryFormat::Any;
+  }
+
+  virtual bool maybe_as_channels_last();
```

Comment:
cc @yf225 Does it cause you problems if these virtual methods are added?

Reply:
these are both big enough features and resolving this issue seems easy enough that each PR should just do the locally optimal thing, IMO.